### PR TITLE
fix gig post

### PR DESCRIPTION
**Pull request by esteban-gs**
Fixes 99% of specs in issue #233 and definitely fixes issue #224 .

- fixes form validations per codeigniter recommendations. (handled by `form_validation.php` in the cofig folder). The reason why issue #224 was happening is because form validation has been handled from the beginning in the form_validation.php file. But someone wrote a new set of validation specs in the `Gig/add` method.
- fixes DB field `CompanyState` (you will need to import the new SQL file to see new updates)
- 1% of issue remaining (to fix 1 &2 we need to implement a more robust ui platform like bootstrap or materialize that have date pickers available):
     1. Date format is not enforced (the temporary fix in this commit is a only suggested date format to the user. The user can still enter any values )
     2. Phone numbers are not enforced for formatting. (currently set to a minimum of 10 chars, and a 
         minimum of 10 chars.
**make sure to run the migration (import SQL file to your DB) to see fixes**
You can see the fix at:  https://www.tegs.tech/gig-central/gig/add

### CODE REVIEW

File: application/config/form_validation.php [hunk 1]
```diff
@@ -18,63 +18,129 @@
  */
 
 // Set form validation rules here.
+
+/*form fileds:
+                //build company info array for model
```
Comment (esteban-gs) [inline application/config/form_validation.php]: Codeigniter allows validation to be passed as a an array. Codeigniter knows which controller and method it applies to, when using this code: `// Set form validation rules here.
$config = array(
        // Sets validation rules for the 'Add a Gig' form
        'gig/add' => array(`
This array gets loaded automatically when the method is called.

File: application/config/form_validation.php [hunk 2]
```diff
@@ -192,5 +258,5 @@
 
 // Sets custom delimeters for error messages on forms.
 // These settings MAY not work if they are above custom validation rules.
-$config['error_prefix'] = '<div class="text-danger">';
-$config['error_suffix'] = '</div>';
\ No newline at end of file
+$config['error_prefix'] = '<span class="badge badge-warning">';
```
Comment (esteban-gs) [inline application/config/form_validation.php]: This adds wrapper tags to the form validation errors globally.

**Comment by esteban-gs** [comment]
@thomaskise , I think I fixed all 4 with a new commit. Please review
1. bootstrap css - we don't want to change this as it could create problems when bootstrap updates are done. Instead, changes shoule be added to the override file pubilic/css/custom_styles.css or pubilic/css/custom_styles_forms.css

- **Fixed. I removed the additions. I simply had copied classes from the newer Bootstwatch versions.**
- **`form_validation.php` has documentation on how to apply styles to form validation messages globaly**

2. SQL file has an issue with these lines of code:
DROP DATABASE IF EXISTS `tegs_tech_gig_central`;
CREATE DATABASE `tegs_tech_gig_central` /*!40100 DEFAULT CHARACTER SET utf8 */;
USE `tegs_tech_gig_central`;

When removed, it works fine.
- **fixed**

3. Another thing is the formatting of the messages themselves has changed. I think it looks fine, but I think we should follow-up with Craig to be sure the change is ok as far as fitting with the rest of the site.
- **related to number 1**
4. The nice to have ... originally the error messages were formatted to specify the the filed name in the message.   Also, the message for the dropdowns was different in that it said "The <field name>" must be selected". I think this looked nicer, but it really should have been using the label names, not the field names. So it would be great to get the messages to updated to say: "The <label name> is reuired" or "The <label name> must be selected"

-  **fixed**

I don't think the change is high priority,  so we could go as is and add another issue to fix this. The first 3 items need to be addressed.

**Comment by thomaskise** [comment]
This looks great now!
Closes task #224

**Pull request merged**